libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigator.java

```diff
@@ -23,6 +23,8 @@ synchronized void updateRoute(String routeJson) {
   }
 
   synchronized NavigationStatus retrieveStatus(Date date) {
+    // We ask for a point slightly in the future to account for lag in location services
+    date.setTime(date.getTime() + 1500);
```

Comment:
What about passing the lag as a parameter of `retrieveStatus` so it can be custom? And then we can add a new option in `MapboxNavigationOptions` with its corresponding default values (constant in `NavigationConstants`).